tests/test_Pretrain.py

```diff
@@ -27,13 +17,13 @@
 }
 
 config2_dict = {
-  "use_theano": True,
+  "use_tensorflow": True,
   "pretrain": "default",
   "num_inputs": 40,
   "num_outputs": 4498,
   "bidirectional": True,
   "hidden_size": (500, 500, 500),
-  "hidden_type": "lstm_opt",
+  "hidden_type": "rec",
```

Comment:
Why is this change needed? Does it actually test whether the layer class exist? Or does it depend on the layer class in any way?

Reply:
lstm_opt is a layer that only exists for Theano, so it will fail for TF.

Reply:
Why does it fail? Why does it actually depend on the layer class?

Reply:
Can you put a normal net dict here instead?

Reply:
For this test it makes sense, for the other not really as it really tests this old legacy format. So I will change it here to a dict and remove the other test.

Reply:
Which do you want to remove? I think all the tests are valid? They just test `get_train_num_epochs` in the end.

Reply:
This does not even make sense, this is literally test 3 then. I will remove both.

Reply:
I think `test_config1` is also important.

Reply:
Yea `test_config2` becomes obsolete then, as it is the same as `test_config3`. but `test_config1` is different and should stay.

Reply:
Did a full rework now.